javascript/MaterialXView/source/viewer.js

```diff
@@ -769,31 +926,108 @@ export class Material
 
         // Update property editor
         const gui = viewer.getEditor().getGUI();
-        this.updateEditor(elem, shader, newMaterial, gui, viewer, closeUI);
+        this.updateEditor(matassign, shader, newMaterial, gui, closeUI, viewer);
 
         if (logDetailedTime)
             console.log("- Per material generate time: ", performance.now() - startGenerateMat, "ms");
 
         return newMaterial;
     }
 
+    clearSoloMaterialUI()
+    {
+        for (let i=0; i<this._materials.length; ++i)
+        {
+            let matassign = this._materials[i];
+            let matUI = matassign.getMaterialUI();
+            if (matUI)
+            {
+                let matTitle = matUI.domElement.getElementsByClassName('title')[0];
+                matTitle.classList.remove('peditor_material_assigned');
+                let img = matTitle.getElementsByTagName('img')[0];    
+                img.src = 'public/shader_ball.svg';
+                //matTitle.classList.remove('peditor_material_unassigned');
+            }
+        }
+    }
+
+    static updateSoloMaterial(viewer, elemPath, materials, event)
+    {
+        // Prevent the event from being passed to parent folder
+        event.stopPropagation();
```

Comment:
This is the only "trick" to capture the "icon" events to not being passed to the parent action which is to collapse the folder.